### DISCUSSION TITLE
Does GotFocus event get fired for a view in a Module that is navigated to?

### DISCUSSION THREAD

**Discussion by stevensrf11** [PrismLibrary/Prism#2398] [WPF]
I have a view which is a UserControl in a Prism Module. This view is navigated to via Prism. I am using the XAML Behaviors eventtrigger to bind the GetFocus event to DelagateCommand in the corresponding view model. The binding connection happening because the output window shows no error on the binding. The problem is the eventrigger is not getting fired. Am I missing something Does a GetFocus event get fired for a view when navigated to? Below is the code.

<UserControl xmlns:dxlc="http://schemas.devexpress.com/winfx/2008/xaml/layoutcontrol"  x:Class="XPub.Admin.Configuration.Module.Views.AdminConfigCreateJob"
             xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
             xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
             xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006" 
             xmlns:d="http://schemas.microsoft.com/expression/blend/2008" 
              xmlns:Behaviors="http://schemas.microsoft.com/xaml/behaviors"
             xmlns:local="clr-namespace:XPub.Admin.Configuration.Module.Views"
              xmlns:prism="http://prismlibrary.com/" 
             xmlns:controls="clr-namespace:XPub.Infrastructure.Controls;assembly=XPub.Infrastructure"
             prism:ViewModelLocator.AutoWireViewModel="True"           
             mc:Ignorable="d" 
             d:DesignHeight="300" d:DesignWidth="600"
            >
    <Behaviors:Interaction.Triggers>
            <Behaviors:EventTrigger EventName="LostFocus">
                <Behaviors:InvokeCommandAction Command="{Binding LostFocusCommand}"   PassEventArgsToCommand="True"/>
            </Behaviors:EventTrigger>
            <Behaviors:EventTrigger EventName="GotFocus">
                <Behaviors:InvokeCommandAction Command="{Binding GotFocusCommand}"   PassEventArgsToCommand="True" />
            </Behaviors:EventTrigger>
        </Behaviors:Interaction.Triggers> 
    <DockPanel>

  public class AdminConfigCreateJobViewModel : ViewModelBase
    {
        #region Fields
        /// <summary>
        /// _xpubServiceAdmin field interface
        /// Contains references to administration services
        /// </summary>
        private readonly IXPubServiceAdmin _xpubServiceAdmin;
        #endregion

        private DelegateCommand<object> _gotFocusCommand;
        public DelegateCommand<object> GotFocusCommand => _gotFocusCommand ?? (_gotFocusCommand = new DelegateCommand<object>(ExecuteGotFocusCommand));

        private DelegateCommand<object> _lostFocusCommand;
        public DelegateCommand<object> LostFocusCommand => _lostFocusCommand ?? (_lostFocusCommand = new DelegateCommand<object>(ExecuteLostFocusCommand));

   
        #region Constructor

        // public AdminConfigCreateJobViewModel()

       public AdminConfigCreateJobViewModel(IXPubServiceAdmin xpubServiceAdmin)
        {
           
            _xpubServiceAdmin = xpubServiceAdmin;
        }
        #endregion

        #region IXPubServiceAdmin Services 
        async void  GetFileRecordInformationAsynv()
        {
            

        }
        #endregion

        private async void ExecuteLostFocusCommand(object eventArgs)
        {
        }

        private async void ExecuteGotFocusCommand(object eventArgs)
        {
            
            var i = 1;

        }

**Comment by brianlagunas** [comment]
Prism has no control over when gotfocus is called in an application.
